File: e-commerce/ShopingApi/ShopingApi/ViewModels/OrderViewModel.cs
namespace ShopingApi.ViewModels
{
    public class OrderViewModel
    {
        public int Id { get; set; }

        public string AspNetUsersId { get; set; }

        public DateTime OrderDate { get; set; }

        public decimal TotalAmount { get; set; } // Match Angular model `TotalAmount`

        public string Note { get; set; } // Order Notes

        public List<OrderItemViewModel> OrderItems { get; set; } // List of items in the order

        public ShippingViewModel ShippingDetails { get; set; } // Shipping details for the order
    }

    public class OrderItemViewModel
    {
        public int ItemId { get; set; }

        public string ItemName { get; set; } // Assuming `Item` has a `Name` property in the `Item` model

        public int Quantity { get; set; }

        public decimal Price { get; set; }
    }

    public class ShippingViewModel
    {
        public string State { get; set; }

        public string PostalCode { get; set; }

        public string Country { get; set; }

        public string Phone { get; set; }

        public string EmailAddress { get; set; }
    }
}

File: e-commerce/ShopApp_ng/src/app/Models/OrderViewModel/orderviewmodel.ts
import { CartViewModel } from "../productViewModel/cart-view-model";

// Assuming CartViewModel and OrderItemViewModel have similar structure.
// If not, update the mapping accordingly.
export class OrderViewModel {
    id: number = 0;
    aspNetUsersId: string = '';
    orderDate: Date = new Date();
    totalAmount: number = 0;
    note: string = '';
    orderItems: OrderItemViewModel[] = [];  // Use OrderItemViewModel instead of CartViewModel
    shippingDetails: ShippingViewModel = new ShippingViewModel();
}

export class OrderItemViewModel {
    itemId: number = 0;
    itemName: string = '';  // Including itemName if needed
    quantity: number = 0;
    price: number = 0;
}

export class ShippingViewModel {
    state: string = '';
    postalCode: string = '';
    country: string = '';
    phone: string = '';
    emailAddress: string = '';
}
